src/main/scala/org/antipathy/scoozie/builder/TransitionBuilder.scala

```diff
@@ -26,35 +24,34 @@ private[scoozie] object TransitionBuilder {
     */
   def build(configSeq: Seq[_ <: Config], yarnConfig: YarnConfig)(implicit credentials: Option[Credentials]): Node = {
-    val actions = Seq(configSeq.map { config =>
-      getType(config) match {
-        case "start"      => NodeWithConfig(Start(), config)
-        case "distcp"     => NodeWithConfig(DistCPAction(config, yarnConfig), config)
-        case "email"      => NodeWithConfig(EmailAction(config), config)
-        case "filesystem" => NodeWithConfig(FsAction(config), config)
-        case "hive"       => NodeWithConfig(HiveAction(config, yarnConfig), config)
-        case "java"       => NodeWithConfig(JavaAction(config, yarnConfig), config)
-        case "pig"        => NodeWithConfig(PigAction(config, yarnConfig), config)
-        case "shell"      => NodeWithConfig(ShellAction(config, yarnConfig), config)
-        case "spark"      => NodeWithConfig(SparkAction(config, yarnConfig), config)
-        case "sqoop"      => NodeWithConfig(SqoopAction(config, yarnConfig), config)
-        case "ssh"        => NodeWithConfig(SshAction(config), config)
-        case "subworflow" => NodeWithConfig(SubWorkflowAction(config, yarnConfig), config)
-        case "end"        => NodeWithConfig(End(), config)
-        case "kill"       => NodeWithConfig(Kill(config.getString("message")), config)
-        //build placeholders for fork,join and decision nodes
-        case "fork"     => NodeWithConfig(Fork(config.getString("name"), Seq.empty), config)
-        case "join"     => NodeWithConfig(Join(config.getString("name"), End()), config)
-        case "decision" => NodeWithConfig(Decision(config.getString("name"), End(), Seq.empty), config)
-        case unknown =>
-          throw new UnknownActionException(s"action type $unknown is invalid")
-      }
+    val actions = Seq(configSeq.map(c => (getType(c), c)).map {
+      case ("start", config)      => NodeWithConfig(Start(), config)
+      case ("distcp", config)     => NodeWithConfig(DistCPAction(config, yarnConfig), config)
+      case ("email", config)      => NodeWithConfig(EmailAction(config), config)
+      case ("filesystem", config) => NodeWithConfig(FsAction(config), config)
+      case ("hive", config)       => NodeWithConfig(HiveAction(config, yarnConfig), config)
+      case ("java", config)       => NodeWithConfig(JavaAction(config, yarnConfig), config)
+      case ("pig", config)        => NodeWithConfig(PigAction(config, yarnConfig), config)
+      case ("shell", config)      => NodeWithConfig(ShellAction(config, yarnConfig), config)
+      case ("spark", config)      => NodeWithConfig(SparkAction(config, yarnConfig), config)
+      case ("sqoop", config)      => NodeWithConfig(SqoopAction(config, yarnConfig), config)
+      case ("ssh", config)        => NodeWithConfig(SshAction(config), config)
+      case ("subworflow", config) => NodeWithConfig(SubWorkflowAction(config, yarnConfig), config)
+      case ("end", config)        => NodeWithConfig(End(), config)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The string literal 'end' appears 2 times in the file.](https://app.codacy.com/app/SimonJPegg/scoozie/pullRequest?prid=3072461)